### ISSUE TITLE
Multiple issues on ARM embedded device

### ISSUE THREAD

**Issue by sectroyer** [hugsy/gef#263]
I used latest version from Git. I use gef natively on a ARM TV with gdb 8.1 and python 2.7 (provided by vendor). I attach text file with the description of issues that I have encountered and "quick fixes" that I applied to overcome the issues (if I managed ;)). I also attach my gef.py. Hope this will be helpful
[issues.txt](https://github.com/hugsy/gef/files/1963942/issues.txt)
[gef.py.zip](https://github.com/hugsy/gef/files/1963943/gef.py.zip)

. Thanks for great tool :)

**Comment by hugsy** [comment]
Hi @sectroyer ,

While I do appreciate the time you spent patching `gef` and giving us this feedback, what you are submitting is totally unexploitable for us since there's no way to reproduce and test correctly before having those changes merged into the `master` of tool so that everyone can enjoy your bug fixes.

So if you are serious about contributing, either send a Pull Request with your changes, or at least fill up correctly the [Issue template](https://github.com/hugsy/gef/blob/master/.github/ISSUE_TEMPLATE.md)  that was created for exactly this reason.

Cheers

**Comment by sectroyer** [comment]
Well I know why you have issue template but I won't be able to provide much more info about this embeded system. Well I know kernel version but doubt that is what you want 😄 Tough if you want I can stick to the template and report 12 issues that I have encountered tough I first wanted you to take a look and judge whether it actually worth reporting. About pull request I doubt my skills are enough to suggest fixes in such a way. What I did were rather quick fixes that probably in some cases do not even fix issues completely. For example I don't think that putting set_arch() in "random" places is a way you would want to fix this issue ? 😄

**Comment by hugsy** [comment]
Hi there,

The problem here (which apparently is more a dead-end than a problem) is that we don't have anything to reproduce and test it. `gef` is constantly tested on several architectures (including ARM) so the fact that you had some changes to provide says that it could be improved (although I have some clues on some of your issues, like 1 & 2 seems to be a bad locale setting, 4 can be fixed by a gef config setting, etc.)

If you can't go any further, maybe a good approach would be to publish this version of `gef` as a fork, detailing the use context of this version (ARM TV etc.).

Cheers

**Comment by sectroyer** [comment]
Tough I mostly use gdb on a tv I just tried it on my pandaboard ES ARM Linux box:
Linux pandaboard 4.15.6 #1 SMP PREEMPT armv7l Generic OMAP4 (Flattened Device Tree) GNU/Linux with Python 2.7.14 
And many of the reported issues are also there. For example 1, there is nothing wrong with locale on my box and still get CPython error. I can sey I didn't need to put set_arch everywhere tough.

So almost all issues can reproduced on two completely different arm boxes 😞 

Issues that I was able to reproduce were:
1, 2, 4, 5, 6, 7, 8, 9, 10, 11, 12 and 13 😞 
So basically only "set_arch" problem is gone.
You want me to open the issue for all of them?
